### DISCUSSION TITLE
Can you support Redis as a messagebus for external messagebus in core-command?

### DISCUSSION THREAD

**Discussion by Pluviophile225** [edgexfoundry/community#97] [Ideas]
Hi, all. We want to use the **redis** messagebus to  communicate with core-command. However the external messagebus for core-command only support MQTT. I think it is quite a necessary demand for customers to choose which kind of messagebus they can use. 
And I find that the internal messagebus has achieved this function. Can you improve the external messagebus as well?

**Comment by cloudxxx8** [comment]
It's possible.  However, it's not in our todo list so far.  We can bring this up in the next release planning meeting, and let TSC to decide the scope.   
 It would helpful if you can provide some complete use cases to help the planning.  Also, it's always welcome to your development contribution.

**Reply by Pluviophile225** [reply]
Our use case is this, we plug EdgeX into the cloud and hope to synchronize content on Kubernetes with content on EdgeX. The solution now is to synchronize content on Kubernetes with content inside EdgeX every once in a while. For devices, each synchronization requires a restful request to get all the command lists, and then a for loop to complete all get requests.
Our optimization now is that we use a cache inside Kubernetes to store everything related to Command, and we want to use MessageBus to initiate Command queries and Get Command requests asynchronously. Because EdgeX uses Redis internally for messagebus, we hope to continue using this microservice. If we deploy an additional microservice for external communication with MQTT, it will cause additional component deployment and waste of resources.
We are more than willing to contribute to the addition of the messagebus implemented by Redis.
